src/neo/Ledger/Blockchain.cs

```diff
@@ -562,6 +565,29 @@ private void Persist(Block block)
             OnPersistCompleted(block);
         }
 
+        private void RecycleRewardGas(SnapshotView snapshot, List<ApplicationExecuted> all_application_executed, List<Tuple<Transaction, long>> recycleRewardGasTxs)
+        {
+            foreach (var tuple in recycleRewardGasTxs)
+            {
+                var tx = tuple.Item1;
+                var recycleRewardGas = tuple.Item2;
+                Script onRecycleRewardGasScript = null;
+                using (ScriptBuilder sb = new ScriptBuilder())
+                {
+                    sb.EmitAppCall(NativeContract.GAS.Hash, "onRecycleRewardGas", tx.Sender, recycleRewardGas);
+                    onRecycleRewardGasScript = sb.ToArray();
+                }
+                using (ApplicationEngine engine = new ApplicationEngine(TriggerType.System, null, snapshot, 0, true))
```

Comment:
If we reuse the snapshot for all iterations, it's better to create only one script, with all the `onRecycleRewardGas` together.

Reply:
It is not all transactions need to be iterated, but only transactions that require recycling need to be iterated. Generally speaking, the number of such transactions should be small, so I don't think it is necessary to merge the scripts into one.

Reply:
I think that it could be faster with only one script execution

Reply:
Let me try to merge them to one script.